flink-tests/src/test/java/org/apache/flink/test/misc/CustomSerializationITCase.java

```diff
@@ -62,127 +68,103 @@ public static Configuration getConfiguration() {
 	}
 
 	@Test
-	public void testIncorrectSerializer1() {
-		try {
-			ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
-			env.setParallelism(PARLLELISM);
-			env.getConfig().disableSysoutLogging();
+	public void testIncorrectSerializer1() throws Exception {
```

Comment:
I think tests testIncorrectSerializer1, testIncorrectSerializer2 and testIncorrectSerializer3  could be deduplicated into
```
<T> void testIncorrectSerializer1(Provider<T> provider)
```
where `T` is either `ConsumesTooMuch` or `ConsumesTooMuchSpanning`

Reply:
yes (hopefully) - well, have a look at the new de-dup code, I tried different implementations and ended up at this one which was most practical

Reply:
I think the following codes can be deduplicated from four tests at least if `T` is not easy to be handled for complete deduplication?

`expectedException.expect(JobExecutionException.class);
expectedException.expectCause(Matchers.both(isA(IOException.class))
	.and(hasProperty("message", containsString("broken serialization"))));

ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
env.setParallelism(PARLLELISM);
env.getConfig().disableSysoutLogging();`